pennylane/transforms/core/transform_dispatcher.py

```diff
@@ -229,6 +234,55 @@ def qfunc_transformed(*args, **kwargs):
 
         return qfunc_transformed
 
+    def _old_device_transform(self, original_device, targs, tkwargs):
+        """Apply the transform on a device"""
+        new_dev = copy.deepcopy(original_device)
+        transform = self._transform
+
+        @new_dev.custom_expand
+        def new_expand_fn(self, tape, *args, **kwargs):  # pylint: disable=unused-variable
+            tapes, _ = transform(tape, *targs, **tkwargs)
+            tape = tapes[0]
+            return self.default_expand_fn(tape, *args, **kwargs)
+
+        return new_dev
+
+    def _device_transform(self, original_device, targs, tkwargs):
+        """Apply the transform on a device"""
+        if self._expand_transform:
+            raise TransformError("Device transform does not support expand transforms.")
+        if self._is_informative:
+            raise TransformError("Device transform does not support informative transforms.")
+        if self._final_transform:
+            raise TransformError("Device transform does not support final transforms.")
+
+        class TransformedDevice(type(original_device)):
+            """A transformed device with updated preprocess method."""
+
+            def __init__(self, original_device, transform):
+                for key, value in original_device.__dict__.items():
+                    self.__setattr__(key, value)
+                self.transform = transform
+                self._original_device = original_device
+
+            def __repr__(self):
+                return f"Transformed Device({original_device.__repr__()} with additional preprocess transform {self.transform})"
+
+            def preprocess(
+                self, config: qml.devices.ExecutionConfig = qml.devices.DefaultExecutionConfig
+            ):
+                """This function updates the original device transform program to be applied."""
+                program, config = self.original_device.preprocess(config)
+                program.push_back(TransformContainer(self.transform, targs, tkwargs))
```

Comment:
was there a particular reason you'd push back instead of insert to front? I haven't thought about whether one is better than the other, just wondering if you had strong opinions for convention here.

Reply:
That's a good question, I think maybe in the future we can add more control on where to put the transform. But for the moment I just follow our LIFO logic from the QNode.